src/nvim/CMakeLists.txt

```diff
@@ -166,28 +176,25 @@ foreach(sfile ${NEOVIM_SOURCES}
     set(f "${d}/${f}")
     set(r "${d}/${r}")
   endif()
-  set(gf1 "${GENERATED_DIR}/${r}.c.generated.h")
-  set(gf2 "${GENERATED_INCLUDES_DIR}/${r}.h.generated.h")
-  set(gf3 "${GENERATED_DIR}/${r}.i")
+  set(gf_c_h "${GENERATED_DIR}/${r}.c.generated.h")
+  set(gf_h_h "${GENERATED_INCLUDES_DIR}/${r}.h.generated.h")
+  set(gf_i "${GENERATED_DIR}/${r}.i")
 
-  if(MSVC)
-    set(PREPROC_OUTPUT /P /Fi${gf3})
-  else()
-    set(PREPROC_OUTPUT -E -o ${gf3})
-  endif()
+  get_preproc_output(PREPROC_OUTPUT ${gf_i})
 
   add_custom_command(
-    OUTPUT "${gf1}" "${gf2}"
+    OUTPUT "${gf_c_h}" "${gf_h_h}"
     COMMAND ${CMAKE_C_COMPILER} ${sfile} ${PREPROC_OUTPUT} ${gen_cflags} ${C_FLAGS_ARRAY}
-    COMMAND "${LUA_PRG}" "${HEADER_GENERATOR}" "${sfile}" "${gf1}" "${gf2}" "${gf3}"
+    COMMAND "${LUA_PRG}" "${HEADER_GENERATOR}" "${sfile}" "${gf_c_h}" "${gf_h_h}" "${gf_i}"
     DEPENDS "${HEADER_GENERATOR}" "${sfile}"
     )
-  list(APPEND NEOVIM_GENERATED_SOURCES "${gf1}")
-  list(APPEND NEOVIM_GENERATED_SOURCES "${gf2}")
+  list(APPEND NEOVIM_GENERATED_SOURCES "${gf_c_h}")
+  list(APPEND NEOVIM_GENERATED_HEADERS "${gf_h_h}")
   if(${d} MATCHES "^api$" AND NOT ${f} MATCHES "^api/helpers.c$")
-    list(APPEND API_HEADERS ${gf2})
+    list(APPEND API_HEADERS ${gf_h_h})
   endif()
 endforeach()
+list(APPEND NEOVIM_GENERATED_SOURCES "${NEOVIM_GENERATED_HEADERS}")
```

Comment:
This is nice I guess. Did not came up with such short form.

BTW, first line of commit messases is advised not to go above 70 unicode code points: you may see github truncating commit summaries above this boundary. Git itself suggests even lesser names: up to 50 display cells because it expects to be optimized for 80-display-cell wide terminals, but our style guide considers this requirement outdated and not aligning well with suggestion “write commit messages like `component: Commit summary`” because `component` part may take too much space out of 50 cells.

(Note: github limits exactly unicode codepoints, I did test this. `git help commit` talks about characters but given how this condition is normally justified on the internet I assume they actually mean “display cells” which only applies to terminals.)